craft_parts/plugins/python_plugin.py

```diff
@@ -154,44 +154,91 @@ def get_build_commands(self) -> List[str]:
         build_commands.append(f"[ -f setup.py ] && pip install {constraints} -U .")
 
         # Now fix shebangs.
+        script_interpreter = self._get_script_interpreter()
         build_commands.append(
             dedent(
                 f"""\
-            find "{self._part_info.part_install_dir}" -type f -executable -print0 | xargs -0 \
-                sed -i "1 s|^#\\!${{PARTS_PYTHON_VENV_INTERP_PATH}}.*$|#\\!/usr/bin/env ${{PARTS_PYTHON_INTERPRETER}}|"
-            """
+                find "{self._part_info.part_install_dir}" -type f -executable -print0 | xargs -0 \\
+                    sed -i "1 s|^#\\!${{PARTS_PYTHON_VENV_INTERP_PATH}}.*$|{script_interpreter}|"
```

Comment:
Is escaping needed for `!` in this case?

Reply:
(I don't know)

Reply:
Maybe not needed, maybe it's there for a reason. I'll leave it there for now.